src/platform/unsupported.rs

```diff
@@ -0,0 +1,36 @@
+use crate::{Error, ErrorKind};
+use log::warn;
+use std::path::PathBuf;
+
+type Result<T> = std::result::Result<T, Error>;
+
+pub struct Handle;
+
+impl Handle {
```

Comment:
I feel like this should actually be a trait if we're going to take this approach.

```rust
trait Handle {
    fn from_fd(fd: RawFd) -> Self;
    fn detach_with_msg(&mut self);
}
```
this way different platforms have different implementations of Handle

the signature of `daemonize` becomes

```rust
fn daemonize<T: Into<PathBuf>>(pid_file: T) -> Result<impl Handle> {}

```

Reply:
Right, I thought so too but the problem is that `RawFd` is a `unix specific` type and then we need to have associated type to cover that. But I prefer to handle that in another PR and merge as it is

Reply:
alright, looks good to me